mcp/src/main/java/io/airlift/mcp/model/JsonRpcErrorCode.java

```diff
@@ -0,0 +1,39 @@
+package io.airlift.mcp.model;
+
+import java.util.Optional;
+
+public enum JsonRpcErrorCode
+{
+    // SDK error codes
+    CONNECTION_CLOSED(-32000),
+    REQUEST_TIMEOUT(-32001),
+
+    // Standard JSON-RPC error codes
+    PARSE_ERROR(-32700),
+    INVALID_REQUEST(-32600),
+    METHOD_NOT_FOUND(-32601),
+    INVALID_PARAMS(-32602),
+    INTERNAL_ERROR(-32603);
+
+    private final int code;
+
+    public int code()
+    {
+        return code;
+    }
+
+    public static Optional<JsonRpcErrorCode> fromCode(int code)
+    {
+        for (JsonRpcErrorCode errorCode : values()) {
```

Comment:
You can create a reverse-index statically initialized:

```
private static Map<Integer, JsonRpcErrorCode> codeMap = Arrays.stream(values())
            .map(errorCode -> Map.entry(errorCode.code, errorCode))
            .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Reply:
done (the `Map.entry` isn't necessary)